File: frontend/src/app/_models/message.ts
import { Genders } from "../_shared/types/genders";

export interface Message {
  id: number;
  senderId: number;
  senderKnownAs: string;
  senderPhotoUrl: string;
  senderGender: Genders;
  recipientId: number;
  recipientKnownAs: string;
  recipientPhotoUrl: string;
  recipientGender: Genders;
  content: string;
  isRead: boolean;
  dateRead: Date;
  messageSent: Date;
}

File: backend/src/DatingApp.Core/Entities/Message.cs
using System;

namespace DatingApp.Core.Entities
{
    public class Message
    {
        public int Id { get; set; }
        public int SenderId { get; set; }
        public virtual User Sender { get; set; }
        public int RecipientId { get; set; }
        public virtual User Recipient { get; set; }
        public string Content { get; set; }
        public bool IsRead { get; set; }
        public DateTimeOffset? DateRead { get; set; }
        public DateTimeOffset MessageSent { get; set; }
        public bool SenderDeleted { get; set; }
        public bool RecipientDeleted { get; set; }
    }
}